File: solutions/basicprogramming1/a.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(0);
  int n, t;
  cin >> n >> t;
  vector<long long> A(n);
  for (long long& a : A) cin >> a;
  if (t == 1) {
    cout << 7 << endl;
  } else if (t == 2) {
    if (A[0] > A[1]) cout << "Bigger" << endl;
    else if (A[0] == A[1]) cout << "Equal" << endl;
    else cout << "Smaller" << endl;
  } else if (t == 3) {
    vector<long long> v{A[0], A[1], A[2]};
    sort(v.begin(), v.end());
    cout << v[1] << endl;
  } else if (t == 4) {
    long long res = 0;
    for (long long a : A) res += a;
    cout << res << endl;
  } else if (t == 5) {
    unsigned long long res = 0;
    for (long long a : A) if (a%2==0) res += a;
    cout << res << endl;
  } else if (t == 6) {
    for (long long a : A) cout << a%26+'a'
    cout << res << endl;
  } else { // t==7
    long long i = A[0];
    unordered_set<long long> vis;
    vis.insert(0);
    while (true) {
      i = A[i];
      if (i >= n) {
        cout << "Out" << endl;
        break;
      }

      if (i+1==n) {
        cout << "Done" << endl;
        break;
      }

      if (vis.count(i)) {
        cout << "Cyclic" << endl;
        break;
      }

      vis.insert(i);
    }
  }
  return 0;
}

File: solutions/basicprogramming1/a.py
n, t = map(int, input().split())
a = [int(x) for x in input().split()]

if t == 1:
    print('7')
elif t == 2:
    print('Bigger') if a[0] > a[1] else print('Equal') if a[0] == a[1] else print('Smaller')
elif t == 3:
    print(sorted(a[:3])[1])
elif t == 4:
    print(sum(a))
elif t == 5:
    print(sum(filter(lambda x: x % 2 == 0, a)))
elif t == 6:
    mod26 = map(lambda x: x % 26, a)
    tochar = map(lambda x: chr(ord('a') + x), mod26)
    print(''.join(tochar))
else:
    i = 0
    vis = {0: None}
    while True:
        i = a[i]
        if i >= n:
            print('Out')
            break
        if i == n - 1:
            print('Done')
            break
        if i in vis:
            print('Cyclic')
            break
        vis[i] = None
